# Useful links

## Cheat sheets

### Read processing/alignment
- [cutadapt](https://cutadapt.readthedocs.io/en/stable/), for trimming reads  
- [STAR aligner](https://github.com/alexdobin/STAR), for aligning reads to a reference  
- [htseq-count](https://htseq.readthedocs.io/en/release_0.11.1/count.html), for quantifying read counts at genomic features  

-----------


### Quality control
- [Picard tools](https://broadinstitute.github.io/picard/), including [CollectRnaSeqMetrics](https://gatk.broadinstitute.org/hc/en-us/articles/360037057492-CollectRnaSeqMetrics-Picard-)
- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/), for basic FASTQ quality control
- [MultiQC](https://multiqc.info/), for aggregating quality control reports

-----------


### Other
- [Conda](https://docs.conda.io/en/latest/)
- [Introduction to RNA-Seq using high-performance computing, workshop by Harvard Chan Bioinformatics Core](https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/schedule/)

-----------


### R programming

- [Base R](./misc/cheat-sheets/base-r.pdf)  
- [Advanced R](./misc/cheat-sheets/advancedR.pdf)  
- [Data import (readr)](./misc/cheat-sheets/data-import.pdf)
- [Data transformation with ggplot2](./misc/cheat-sheets/data-transformation.pdf)  
- [Data visualization](./misc/cheat-sheets/data-visualization-2.1.pdf)  
- [Regular expressions](./misc/cheat-sheets/regex.pdf)
- [Strings](./misc/cheat-sheets/strings.pdf)  


-----------


### RStudio

- [R Studio](./misc/cheat-sheets/rstudio-ide.pdf)  
- [R markdown](./misc/cheat-sheets/rmarkdown-2.0.pdf)  

-----------

## Differential expression analysis tools

- [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html), original manuscript: [Love et al, 2014](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8)  
- [tximport](https://bioconductor.org/packages/release/bioc/html/tximport.html), original manuscript: [Soneson et al, 2014](https://f1000research.com/articles/4-1521/v1) 
- [edgeR](https://www.bioconductor.org/packages/release/bioc/html/edgeR.html), original manuscript: [Robinson et al, 2010](https://academic.oup.com/bioinformatics/article/26/1/139/182458)  
- [limma](http://bioconductor.org/packages/release/bioc/html/limma.html), original manuscript: [Ritchie et al, 2015](https://academic.oup.com/nar/article/43/7/e47/2414268)  
- [limma-voom](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29)  

-----------

## Differential expression method comparisons

- [Comparison of software packages for detecting differential expression in RNA-seq studies](https://bioconductor.org/packages/release/bioc/html/DESeq2.html). Seyednasrollah, *et al* 2013, *Genome Biol*. 
- [A comparison of methods for differential expression analysis of RNA-seq data](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-91). Soneson, *et al* 2013, *BMC Bioinfo.*. 
- [Comprehensive evaluation of differential gene expression analysis methods for RNA-seq data](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-9-r95). Rapaport, *et al* 2013, *Genome Biol*. 
- [Systematic comparison and assessment of RNAâ€‘seq procedures for gene expression quantitative analysis](https://www.nature.com/articles/s41598-020-76881-x). Corchete, *et al* 2020, *Scientific Reports*. 
